src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotification.java

```diff
@@ -182,7 +184,10 @@ private static GitHub lookUpGitHub(@NonNull Job<?, ?> job) throws IOException {
                 return Connector.connect(
                         source.getApiUri(),
                         Connector.lookupScanCredentials(
-                                job, source.getApiUri(), source.getScanCredentialsId(), source.getRepoOwner()));
+                                source.getOwner(),
```

Comment:
The changes here are kind of a bug fix. Essentially we make sure to pass the `SCMSource` owner instead of the job itself so that `Connector.lookupScanCredentials` correctly identifies this as a trusted context and infers the owner and repository correctly.